core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -1645,6 +1691,95 @@ object Parser extends ParserInstances {
       }
     }
 
+    final def replicateCapture[A, B](
+        p: Parser[A],
+        n: Int,
+        state: State,
+        append: Appender[A, B]
+    ): Boolean = {
+      var offset = state.offset
+      var cnt = 0
+      var keepParsing = true
+      var result = false
+
+      while (keepParsing) {
+        val a = p.parseMut(state)
+        if ((state.error eq null) && cnt < n) {
+          cnt += 1
+          append.append(a)
+          offset = state.offset
+        } else {
+          keepParsing = false
+          // there has been an error
+          if (cnt == n) {
+            // we correctly read n items
+            // reset the error to make the success
+            state.error = null
+            result = true
+          } else {
+            // we didn't read n items
+            result = false
+          }
+        }
+      }
+
+      result
+    }
+
+    final def replicateNoCapture[A](p: Parser[A], n: Int, state: State): Unit = {
+      var offset = state.offset
+      var cnt = 0
+      var keepParsing = true
+
+      while (keepParsing) {
+        p.parseMut(state)
+        if ((state.error eq null) && cnt < n) {
+          cnt += 1
+          offset = state.offset
+        } else {
+          keepParsing = false
+          // there has been an error
+          if (cnt == n) {
+            // we correctly read n times
+            // reset the error to make the success
+            state.error = null
+          }
+        }
+      }
+    }
+
+    case class Replicate[A, B](p1: Parser[A], n: Int, acc: Accumulator[A, B]) extends Parser[B] {
+      private[this] val ignore: B = null.asInstanceOf[B]
+
+      override def parseMut(state: State): B = {
+        if (state.capture) {
+          val app = acc.newAppender()
+          if (replicateCapture(p1, n, state, app)) app.finish()
+          else ignore
+        } else {
+          replicateNoCapture(p1, n, state)
+          ignore
+        }
+      }
+    }
+
+    case class Replicate1[A, B](p1: Parser1[A], n: Int, acc: Accumulator1[A, B])
```

Comment:
What do you think of this idea instead: we take a min and a max in Rep, and for replicate then min is the max. But this also enables a new combinator: parse at most n things. Of course that is a bit of a weird combinator so maybe not with special casing.